src/components/checkbox/tests/GroupableCheckboxReducers.spec.ts

```diff
@@ -0,0 +1,299 @@
+        expect(groupableCheckboxsState.length).toBe(1, 'parent checkbox was not added in the new state');
+        if (groupableCheckboxsState.length) {
+          expect(groupableCheckboxsState[0].parent).toBeDefined();
+          expect(groupableCheckboxsState[0].parentId).toBe('parentId');
+        }
+      });
+
+      describe('with a existing state', () => {
+        let currentState: IGroupableCheckboxesState[];
+
+        beforeEach(() => {
+          currentState = [{
+            total: 1,
+            nbChecked: 0,
+            parentId,
+            checkboxes: [
+              {
+                id: 'id1',
+                checked: false,
+              },
+            ],
+          },
+          ];
+        });
+
+        it('should add the parent in the current state', () => {
+          action = addGroupedCheckbox(parentId, false, undefined, true);
+          const groupableCheckboxsState: IGroupableCheckboxesState[] = groupableCheckboxesReducer(currentState, action);
+          expect(groupableCheckboxsState.length).toBe(1, 'parent checkbox was not added in the current state');
+          if (groupableCheckboxsState.length) {
+            expect(groupableCheckboxsState[0].parent).toBeDefined();
+            expect(groupableCheckboxsState[0].parentId).toBe('parentId');
+          }
+        });
+
+        it('should add the child checkbox in the current state', () => {
+          action = addGroupedCheckbox('id', false, parentId);
+          const groupableCheckboxsState: IGroupableCheckboxesState[] = groupableCheckboxesReducer(currentState, action);
+          expect(groupableCheckboxsState.length).toBe(1, 'child checkbox was not added in the current state');
+          if (groupableCheckboxsState.length) {
+            expect(groupableCheckboxsState[0].checkboxes.length).toBe(2);
+          }
+        });
+
+        it('should update the total on add child checkbox with an existing state', () => {
+          action = addGroupedCheckbox('id', false, parentId);
+          const groupableCheckboxsState: IGroupableCheckboxesState[] = groupableCheckboxesReducer(currentState, action);
+          expect(groupableCheckboxsState.length).toBe(1, 'child checkbox was not added in the new state');
+          if (groupableCheckboxsState.length) {
+            expect(groupableCheckboxsState[0].total).toBe(2, 'total was not updated');
+          }
+        });
+
+        it('should have nbChecked equal to 1 on add checkbox checked with an existing state', () => {
+          action = addGroupedCheckbox('id', true, parentId);
+          const groupableCheckboxsState: IGroupableCheckboxesState[] = groupableCheckboxesReducer(currentState, action);
+          expect(groupableCheckboxsState.length).toBe(1, 'child checkbox was not added in the new state');
+          if (groupableCheckboxsState.length) {
+            expect(groupableCheckboxsState[0].nbChecked).toBe(1, 'nbChecked has been updated');
+          }
+        });
+      });
```

Comment:
You should add a few tests to make sure the state being returned is a new object when it is changing.

Reply:
Added